app/services/importers/okc_blotter/pdf.rb

```diff
@@ -0,0 +1,139 @@
+        return @json if response.success?
+
+        raise StandardError.new "Error parsing pdf: #{@json['message']}"
+      end
+
+      def pdf
+        return @pdf if @pdf
+        pdf = Bucket.new.get_object("#{s3_path}/#{@date}.pdf").body rescue false
+        @pdf = pdf if pdf
+        pdf
+      end
+
+      def build_bookings(booking_hash)
+        ::OkcBlotter::Booking.new({
+                                    dob: booking_hash['dob'],
+                                    sex: booking_hash['sex'],
+                                    last_name: booking_hash['lastName'],
+                                    first_name: booking_hash['firstName'],
+                                    race: booking_hash['race'],
+                                    booking_number: booking_hash['bookingNumber'],
+                                    booking_date: booking_hash['bookingDate'],
+                                    inmate_number: booking_hash['inmateNumber'],
+                                    transient: booking_hash['transient'],
+                                    zip: booking_hash['zip'],
+                                    booking_type: booking_hash['bookingType'],
+                                    offenses: booking_hash['offenses'].map do |offense_hash|
+                                      build_offense(offense_hash)
+                                    end
+                                  })
+      end
+
+      def build_offense(offense_hash)
+        ::OkcBlotter::Offense.new({
+                                    type: offense_hash['type'],
+                                    bond: offense_hash['bond'],
+                                    code: offense_hash['code'],
+                                    dispo: offense_hash['dispo'],
+                                    charge: offense_hash['charge'],
+                                    warrant_number: offense_hash['warrantNumber'],
+                                    citation_number: offense_hash['citationNumber']
+                                  })
+      end
+
+      def s3_path
+        'okc_blotter'
+      end
+
+      def url
+        'https://okc-blotter.herokuapp.com'
+      end
+
+      def auth_token
+        ENV['OKC_BLOTTER_AUTH_TOKEN']
+      end
+
+      def self.import_since_last_run
+        last_run_date = ::OkcBlotter::Pdf.order(date: :desc).first.date
+        (last_run_date + 1.day..DateTime.now.to_date).each do |date|
+          begin
+            perform(date)
+          rescue StandardError => error
+            puts "error processing #{date}. Data not saved" #todo: log these somewhere?
+            puts "error was: #{error}"
+          end
+        end
+      end
+
+      # Note: PDFs only are on the website for 30 days.
+      # If the pdfs have already been saved to s3 you may be able to parse longer
+      def self.import_missing_dates(since_date = (DateTime.now - 1.month).to_date)
+        query = <<-sql
```

Comment:
performs a sql query to find missing dates (since this importer should get a pdf every day?) then  runs the perform method using these dates?